hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/TestContainerReportHandling.java

```diff
@@ -97,6 +95,19 @@ void testDeletingOrDeletedContainerTransitionsToClosedWhenNonEmptyReplicaIsRepor
         ContainerID containerID = ContainerID.valueOf(keyLocation.getContainerID());
         waitForContainerClose(cluster, containerID.getId());
 
+        // also wait till the container is closed in SCM
+        GenericTestUtils.waitFor(() -> {
+          ContainerManager containerManager = cluster.getStorageContainerManager().getContainerManager();
+          try {
+            if (containerManager.getContainer(containerID).getState() != HddsProtos.LifeCycleState.CLOSED) {
+              return false;
+            }
+          } catch (ContainerNotFoundException e) {
+            return false;
+          }
+          return true;
+        }, 2000, 20000);
+
```

Comment:
I guess at this point SCM should already have the container, `ContainerNotFoundException` can be considered a problem.  If so, we could reuse the check (extracted to a method in `TestHelper`) from the test's end:

https://github.com/apache/ozone/blob/6648965b5ce224cff535fd3bf633fcd251c54167/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/TestContainerReportHandling.java#L125-L132